client/src/app/pages/external/jira/instance-form.tsx

```diff
@@ -0,0 +1,292 @@
+  onClose,
+}) => {
+  const { t } = useTranslation();
+
+  const [axiosError, setAxiosError] = useState<AxiosError>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [insecureCommunication, setInsecureCommunication] = useState(false);
+
+  const { jiraTrackers: instances } = useFetchJiraTrackers();
+  const { identities } = useFetchIdentities();
+
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const onCreateInstanceSuccess = (_: AxiosResponse<JiraTracker>) =>
+    pushNotification({
+      title: t("terms.jiraInstanceCreated"),
+      variant: "success",
+    });
+
+  const onCreateUpdateInstanceError = (error: AxiosError) => {
+    setAxiosError(error);
+  };
+
+  const { mutate: createInstance } = useCreateJiraTrackerMutation(
+    onCreateInstanceSuccess,
+    onCreateUpdateInstanceError
+  );
+
+  const onUpdateInstanceSuccess = (_: AxiosResponse<JiraTracker>) =>
+    pushNotification({
+      title: t("terms.jiraInstanceUpdated"),
+      variant: "success",
+    });
+
+  const { mutate: updateInstance } = useUpdateJiraTrackerMutation(
+    onUpdateInstanceSuccess,
+    onCreateUpdateInstanceError
+  );
+
+  const onSubmit = (formValues: FormValues) => {
+    const matchingCredential = identities.find(
+      (identity) => formValues?.credentials === identity.name
+    );
+
+    const payload: JiraTracker = {
+      name: formValues.name.trim(),
+      url: formValues.url.trim(),
+      id: formValues.id,
+      kind: formValues.kind,
+      metadata: { projects: [] },
+      message: "",
+      connected: false,
+      identity: matchingCredential
+        ? { id: matchingCredential.id, name: matchingCredential.name }
+        : undefined,
+    };
+    if (instance) {
+      updateInstance({
+        ...payload,
+      });
+    } else {
+      createInstance(payload);
+    }
+    onClose();
+  };
+
+  const validationSchema: yup.SchemaOf<FormValues> = yup.object().shape({
+    id: yup.number().defined(),
+    name: yup
+      .string()
+      .trim()
+      .required(t("validation.required"))
+      .min(3, t("validation.minLength", { length: 3 }))
+      .max(120, t("validation.maxLength", { length: 120 }))
+      .test(
+        "Duplicate name",
+        "An identity with this name already exists. Use a different name.",
+        (value) => duplicateNameCheck(instances, instance || null, value || "")
+      ),
+    url: yup
+      .string()
+      .trim()
+      .required(t("validation.required"))
+      .max(250, t("validation.maxLength", { length: 250 })),
+    kind: yup.mixed<JiraKind>().required(),
+    credentials: yup.string().required(),
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting, isValidating, isValid, isDirty },
+    getValues,
+    setValue,
+    control,
+    watch,
+  } = useForm<FormValues>({
+    defaultValues: {
+      name: instance?.name || "",
+      url: instance?.url || "",
+      id: instance?.id || 0,
+      kind: instance?.kind || undefined,
```

Comment:
```suggestion
      kind: instance?.kind,
```
Undefined will be assigned if kind does not exist on the instance object. That is wha optional chaining gets you.